src/docfx/build/page/BuildPage.cs

```diff
@@ -12,6 +12,8 @@ namespace Microsoft.Docs.Build
 {
     internal static class BuildPage
     {
+        private static readonly object s_lock = new object();
```

Comment:
consider let `Output` manage per output path locks, to ensure outputs to the same outputPath is saved sequentially.

Reply:
I think `context.PublishModelBuilder.TryAdd` and `context.Output.Write` should be an atom operation here.
Let's say, we have `redirection file` A and `source file` B with the same output.
A and B are building simultaneously, B call `TryAdd` firstly and then `write`, A call `TryAdd` later and should rewrite the previous `write` from B. If `TryAdd` and `write` are different operations, it could happen that the later B rewrite finishes before A write.

Reply:
Right, the current implementation does not achieve this either. I'm thinking about sth. like below where the lock is hidden inside a disposable:

```csharp
using (publishManifestBuilder.TryAdd(publishItem, out var writeOutput))
{
    if (writeOutput)
    {
        // write output
    }
}
```

Reply:
The case I mentioned above happened during our UT, it is fixed now. Why the current implementation does not achieve this either?

Reply:
The return value of `publishManifestBuilder.TryAdd` is the same as `out var writeOutput`